src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -50,30 +66,88 @@ export default class SqlX extends SelfServeBaseClass {
   };
 
   public onSave = async (currentValues: Map<string, SmartUiInput>): Promise<SelfServeNotification> => {
-    validate(currentValues);
-    // TODO: add pre processing logic before calling the updateDedicatedGatewayProvisioning() RP call.
-    throw new Error(`onSave not implemented. No. of properties to save: ${currentValues.size}`);
+    const response = await getCurrentProvisioningState();
```

Comment:
why call this here again? it is called once in initialize right? save that state and go from there

Reply:
Portal might not be the only way to provision and deprovision dedicated gateway correct?